### PR TITLE
2017 5TeV cards

### PR DESCRIPTION
**Pull request by gkrintir**
These are the powheg & madgraph5_amc@nlo cards for 5.02 TeV production.

All powheg cards successfully tested in CMSSW_9_3_0.

All madgraph5_amc@nlo cards tested to reproduce the results from the cards used in 2015 5.02 TeV production (within unc.).

Additional cards may come later (e.g. follow up on WZ&ZZ production currently known to be buggy).

### CODE REVIEW

File: bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/dyellell012j_5f_NLO_FXFX/dyellell012j_5f_NLO_FXFX_proc_card.dat [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import model loop_sm-ckm_no_b_mass
+#switch to diagonal ckm matrix if needed for speed
+#import model loop_sm-no_b_mass
```
Comment (perrozzi) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/dyellell012j_5f_NLO_FXFX/dyellell012j_5f_NLO_FXFX_proc_card.dat]: please follow these instructions to cleanup the datacards
https://twiki.cern.ch/twiki/bin/view/CMS/Moriond18MC#Checks_to_the_datacards_BEFORE_T

Comment (gkrintir) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/dyellell012j_5f_NLO_FXFX/dyellell012j_5f_NLO_FXFX_proc_card.dat]: Sorry that skipped my attention Luca. I removed the commented/unuseful lines from both madgraph proc cards.

Comment (perrozzi) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/5TeV/dyellell012j_5f_NLO_FXFX/dyellell012j_5f_NLO_FXFX_proc_card.dat]: remove p and j definitions

**Comment by efeyazgan** [comment]
I think there are other commented inputs that you can get rid of like:
#set BW_cut 15                # cut on how far the particle can be off-shell
# 
#define ell+ = e+ mu+ ta+
#define ell- = e- mu- ta-
# specify the decay for the final state particles
#decay w+ > ell+ vl
#decay w- > ell- vl~
and there are others...

**Comment by gkrintir** [comment]
Thanks Efe and Luca. Done.

**Comment by efeyazgan** [comment]
In the powheg cards there are commented inputs to be removed as well

**Comment by gkrintir** [comment]
Done. 
Corrected also the comment on decay mode: 20000    both top go into **e** l nu

File: bin/Powheg/production/2017/5TeV/WWTo2L2Nu_NNPDF30_5TeV/WWTo2L2Nu_NNPDF31_5TeV.input~ [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                   ! seeds in the same directory.
+                   ! If set to 1, the program asks for an integer j;
+                   ! The file pwgseeds.dat at line j is read, and the
+                   ! integer at line j is used to initialize the random
+                   ! sequence for the generation of the event.
+                   ! The event file is called pwgevents-'j'.lhe
+
+
+
+#dronly 0        ! default 0; if 1 only double resonant diagrams
```
Comment (efeyazgan) [inline bin/Powheg/production/2017/5TeV/WWTo2L2Nu_NNPDF30_5TeV/WWTo2L2Nu_NNPDF31_5TeV.input~]: please delete this line

File: bin/Powheg/production/2017/5TeV/WWTo2L2Nu_NNPDF30_5TeV/WWTo2L2Nu_NNPDF31_5TeV.input~ [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+#zerowidth 1     ! default 0 (offshell Zs); if 1 generate Z's on shell
```
Comment (efeyazgan) [inline bin/Powheg/production/2017/5TeV/WWTo2L2Nu_NNPDF30_5TeV/WWTo2L2Nu_NNPDF31_5TeV.input~]: please delete this line

File: bin/Powheg/production/2017/5TeV/WWTo2L2Nu_NNPDF30_5TeV/WWTo2L2Nu_NNPDF31_5TeV.input~ [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+pdfreweight 1       ! PDF reweighting
```
Comment (efeyazgan) [inline bin/Powheg/production/2017/5TeV/WWTo2L2Nu_NNPDF30_5TeV/WWTo2L2Nu_NNPDF31_5TeV.input~]: why not 0?

File: bin/Powheg/production/2017/5TeV/WWTo2L2Nu_NNPDF30_5TeV/WWTo2L2Nu_NNPDF31_5TeV.input~ [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+storeinfo_rwgt 1
```
Comment (efeyazgan) [inline bin/Powheg/production/2017/5TeV/WWTo2L2Nu_NNPDF30_5TeV/WWTo2L2Nu_NNPDF31_5TeV.input~]: why not 0?

File: bin/Powheg/production/2017/5TeV/WWTo2L2Nu_NNPDF30_5TeV/WWTo2L2Nu_NNPDF31_5TeV.input~ [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+withnegweights 0
```
Comment (efeyazgan) [inline bin/Powheg/production/2017/5TeV/WWTo2L2Nu_NNPDF30_5TeV/WWTo2L2Nu_NNPDF31_5TeV.input~]: why 0?

**Comment by gkrintir** [comment]
Hi @efeyazgan please note that these are the deleted cards (~). I forgot to exclude them from git config.

**Comment by efeyazgan** [comment]
In the powheg cards, CKM matrix elements needs to be updated following: https://github.com/cms-sw/genproductions/blob/master/bin/Powheg/production/2017/13TeV/st_tch_4f_hadr_ckm_13TeV/st_tch_4f_hadr_ckm_top_13TeV.input (however I see that yours are the same as https://github.com/cms-sw/genproductions/blob/master/bin/Powheg/production/2017/13TeV/st_tW_5f_dr_ckm_13TeV/ST_tW_DR_5f_ckm_antitop_13TeV.input) @ckmackay @jfernan2

**Comment by efeyazgan** [comment]
Right sorry, but the question on ckm is still relevant.

**Comment by gkrintir** [comment]
Oh good point, thx a lot for spotting. 

What is the input source for the values in st_tch_4f_hadr_ckm_top_13TeV card ?

**Comment by efeyazgan** [comment]
If I remember correctly, they were from PDG.

**Comment by gkrintir** [comment]
Hi @efeyazgan sure. But if we are using the best fit values, I was wondering from which PDG version these values were taken from (I couldn't figure out). 
Shall we start adding comments next to the values to specify for any future reference ? 
Also let's start using values for all parameters (CKM & EW) consistent with the PDG version, right ?

Thanks again for your nice comment.

**Comment by efeyazgan** [comment]
If I remember correctly @Andrej-CMS added the new values.

**Comment by Andrej-CMS** [comment]
Dear all,
I changed the CKM values for ST tch 4FS  in https://github.com/cms-sw/genproductions/pull/1332 .

The values were taken from the 2016 PDG [1], since it was the latest version that was available at that time.

[1] http://pdg.lbl.gov/2017/reviews/rpp2016-rev-ckm-matrix.pdf

**Comment by efeyazgan** [comment]
Thanks. Please make sure other parameters are also consistent.

**Comment by gkrintir** [comment]
Hi @Andrej-CMS @efeyazgan thx for the answer. 

But I suggest taking these values from global fits and not from the individual, yet the most sensitive, measurements. 

So for 

- CKM matrix elements means: Table 12.27 from [PDG17](http://pdg.lbl.gov/2017/reviews/rpp2017-rev-ckm-matrix.pdf#page=15)

- EW parameters means: Table 10.4 column SM from [PDG17](http://pdg.lbl.gov/2017/reviews/rpp2017-rev-standard-model.pdf#page=28)

**Comment by efeyazgan** [comment]
Yes, that is probably better but please discuss this with TopModGen and let us know the decision.

**Comment by gkrintir** [comment]
Hi @efeyazgan  all values in ST & TT powheg cards consistently updated from PDG 17 review

**Comment by jfernan2** [comment]
I'd like to revive this PR since (I hope) we are still on time (samples not submitted yet): when I was asked about this last week, I could not look into details the changes since I was offline. 

I realize now that the values are not Global Fits but an average of the experimental measurements (we checked this explicitly for W and top width).

Moreover, the charm and b quark masses are in MSbar scheme now with this PR. We think the MC generators (at least Powheg) need the pole mass as input, and this is a real change: bmass =4.8 GeV (pole mass) used up to now, bmass=4.18 GeV (MSbar scheme) used now in 5TeV [1.5 and 1.28 for cmass resp.]

We think this should be revisited, at least for the bmass, cmass, and probably W and top widths.
Thanks

**Comment by gkrintir** [comment]
Hi @jfernan2 i was referring to global fit for the CKM elements.

Exactly, for the particle widths are avgs. I think this is a proper way to read these values from, but let me know if you have a better alternative.

Thx a lot for spotting the msbar vs pole mass issue. Still the old values are somehow not consistent with the pole mass values in PDG17 review, i.e.,  cmass=1.67 [1] & bmass=4.78 [2].

[1] http://pdg.lbl.gov/2017/listings/rpp2017-list-c-quark.pdf
[2] http://pdg.lbl.gov/2017/listings/rpp2017-list-b-quark.pdf

**Comment by jfernan2** [comment]
Sorry for the late answer, I have been consulting with TOP PAG.

For the b/c masses I'd stick to pole/MC values, as in mtop.

For top width: I don't agree this is a more precise value, since MC goes in different way (mtop is 172.5 at MC) and the measurements are biased due to tensions between LHC and Tevatron, besides disagreements in the definition. 
Top width has even more tension between CMS measurement and D0 (poor measurement):
http://pdglive.lbl.gov/DataBlock.action?node=Q007W&init=0

Up to now we have been using a PDG formula: +-2%/1GeV starting from a 1.31 value at mtop=172.5GeV so,since you plan to produce mtop=161 and mtop=178 samples, I would suggest you stick to this formula to have consistent values with CMS measurements and previous Run1/2 analysis.

W width, the value used up to now was more in line of LEP measurements, while current average has a bigger Tevatron weight and less uncertainty. So, I'm fine with that change, but we should propagate it to future MC productions.

CKM values and for the rest of parameters, the changes are small or justified, so I'm fine with them, but they should be propagated to future productions too.

We are happy to discuss this in anty of TOP TMG meetings if you want.
Thanks for the update!

**Comment by gkrintir** [comment]
Thanks @jfernan2 Sure I can gather the values and present during the next TOP TMG meeting.

**Comment by jfernan2** [comment]
Great! Can you make it on tomorrow's TMG?
https://indico.cern.ch/event/688821/

**Comment by efeyazgan** [comment]
I won't be able to attend the meeting. It would be great if you could write the summary and conclusions here. Thanks.

**Comment by gkrintir** [comment]
Hi @efeyazgan Two main actions:

- for 5.02 TeV cards: the shown values will be implemented
- the 5.02 TeV cards could be then a 'baseline' for all CMS wide future productions. This means the discussion shall be brought in GEN and further make a twiki where we can keep up to date the input values (Rebeca mentioned there used to be one). Also, once agreed, the proposed changes shall be propagated to MadGraph.

@jfernan2  can further add smth I missed ;D

**Comment by jfernan2** [comment]
Hi Efe, (sorry Georgios, I was writing when your comment appeared, so I decide to leave it as it is even if I repeat your summary, thanks) 
we decided for 5TeV:
1) mtop, stick/remain to 172.5
2) top width: stick to the current 1.33GeV for 172.5 and the PDG formula of +-2% / GeV of mtop mass
3) stick to mbottom/mcharm pole mass values
4) update mW and W width to the latest most precise measurements/averages as he proposes (wmass 80.385  vs current wmass 80.4, wwidth 2.085 vs current wwidth 2.141)
5) update latest CKM or parameter values to the latest (2017) PDG values since differences are small and latest values have less uncertainty.

We will adopt 4) and 5) points in future TOP MC productions for 2018 data. We would like to make this a general policy in CMS to be adopted by all PAGs&POGs MC contacts in any SM sample. We will need a presentation to show the proposal in a 2018 GEN meeting and place in the GEN twikis to include a baseline. There should be a check for this in future (2018) datacards merged in git
Thanks

**Comment by efeyazgan** [comment]
Thanks a lot @jfernan2 and @gkrintir 
The proposal seems good and we took note of the talk for the proposal. We'll ping you in January.

**Comment by gkrintir** [comment]
btw #1600 takes care of that

Regards and thanks

**Pull request merged**